raft/src/main/java/org/apache/kafka/raft/internals/RequestSender.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.raft.internals;
+
+import org.apache.kafka.common.Node;
+import org.apache.kafka.common.network.ListenerName;
+import org.apache.kafka.common.protocol.ApiMessage;
+
+import java.util.OptionalLong;
+import java.util.function.Supplier;
+
+/**
+ * Interface for sending KRaft requests.
+ *
+ * Responsible for managing the connection state and sending request when the connection is
+ * available.
+ */
+interface RequestSender {
```

Comment:
I'm really late to the party. But why is this interface package protected instead of public?
The interface implementing it is public (`DefaultRequestSender`) and it's used as a parameter in `AddVoterHandler`'s contructor (which is a public class and constructor). This causes a leak of a package protected class outside of its scope.